### PR TITLE
Split 835b488

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/event_scripts.h [hunk 1]
```diff
@@ -1186,4 +1186,6 @@ extern const u8 Text_1BCA95[];
 extern const u8 Text_1BCACB[];
 extern const u8 Text_1BCAF2[];
```
Comment (jiangzhengwenjz) [inline include/event_scripts.h]: text DECLs shouldn't be here

File: src/wild_pokemon_area.c [hunk 2]
```diff
@@ -265,7 +265,7 @@ static bool32 PokemonInAnyEncounterTableInMap(const struct WildPokemonHeader * d
         return TRUE;
     if (PokemonInEncounterTable(data->waterMonsInfo, species, 5))
         return TRUE;
-    if (PokemonInEncounterTable(data->fishingMonsInfo, species, 12))
+    if (PokemonInEncounterTable(data->fishingMonsInfo, species, 12)) // 10
```
Comment (jiangzhengwenjz) [inline src/wild_pokemon_area.c]: What does this comment mean?

File: src/wild_encounter.c [hunk 3]
```diff
@@ -0,0 +1,748 @@
...
+
+extern const u8 gUnknown_83CA71C[][12];
+
+static bool8 UnlockedTanobyOrAreNotInTanoby(void);
+static u32 GenerateUnownPersonalityByLetter(u8 letter);
+static bool8 IsWildLevelAllowedByRepel(u8 level);
+static void ApplyFluteEncounterRateMod(u32 *rate);
+static u8 GetFluteEncounterRateModType(void);
+static void ApplyCleanseTagEncounterRateMod(u32 *rate);
+static u8 IsLeadMonHoldingCleanseTag(void);
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: bool8

File: src/wild_encounter.c [hunk 4]
```diff
@@ -0,0 +1,748 @@
...
+    return 4;
+}
+
+static u8 ChooseWildMonIndex_Fishing(u8 rod)
+{
+    u8 slot = 0;
+    u8 pct = Random() % 100;
+    switch (rod)
+    {
+    case 0: // old
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: Is it better to update case 1 and case 2 with their meanings?

File: src/wild_encounter.c [hunk 5]
```diff
@@ -0,0 +1,748 @@
...
+    } while (GetUnownLetterByPersonalityLoByte(personality) != letter);
+    return personality;
+}
+
+u8 GetUnownLetterByPersonalityLoByte(u32 personality)
+{
+    return (((personality & 0x3000000) >> 18) | ((personality & 0x30000) >> 12) | ((personality & 0x300) >> 6) | (personality & 0x3)) % 0x1C;
+}
+
+static bool8 TryGenerateWildMon(const struct WildPokemonInfo * info, u8 tableIdx, u8 a2)
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: a2 might be bool8 as it's set to 0 only when in sweetscent encounter (else always 1)

File: src/wild_encounter.c [hunk 6]
```diff
@@ -0,0 +1,748 @@
...
+{
+    if ((Random() % 100) >= 60)
+        return FALSE;
+    return TRUE;
+}
+
+bool8 StandardWildEncounter(u32 currMetaTileBehavior, u16 previousMetaTileBehavior)
+{
+    u16 headerId;
+    struct Roamer *roamer;
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: Inconsistent pointer space style

File: src/wild_encounter.c [hunk 7]
```diff
@@ -0,0 +1,748 @@
...
+}
+
+static bool8 DoGlobalWildEncounterDiceRoll(void)
+{
+    if ((Random() % 100) >= 60)
+        return FALSE;
+    return TRUE;
+}
+
+bool8 StandardWildEncounter(u32 currMetaTileBehavior, u16 previousMetaTileBehavior)
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: metatile is one word

File: src/wild_encounter.c [hunk 8]
```diff
@@ -0,0 +1,748 @@
...
+    encRate /= 256;
+    if (sWildEncounterData.stepsSinceLastEncounter >= minSteps)
+        return TRUE;
+    sWildEncounterData.stepsSinceLastEncounter++;
+    if ((Random() % 100) < encRate)
+        return TRUE;
+    return FALSE;
+}
+
+bool8 TryStandardWildEncounter(u32 currMetaTileBehavior)
```
Comment (jiangzhengwenjz) [inline src/wild_encounter.c]: metatile is one word

File: src/wild_pokemon_area.c (hunk 2 above)
Comment (PikalaxALT) [reply src/wild_pokemon_area.c]: It's supposed to be 10 but the developer mistakenly put 12 here. This causes the game to think there's two entries more than there actually are when looking up a Pokemon's nest. By dumb luck, these two entries are 0 padding and the high short of the pointer to the fishing encounters array, neither of which is a valid species identifier, so the player sees no ill effect.

File: src/wild_encounter.c (hunk 8 above)
Comment (PikalaxALT) [reply src/wild_encounter.c]: tell that to emerald

File: src/wild_encounter.c (hunk 5 above)
Comment (PikalaxALT) [reply src/wild_encounter.c]: Nope, it's the check flags from emerald.

File: src/wild_encounter.c (hunk 4 above)
Comment (PikalaxALT) [reply src/wild_encounter.c]: I'll do you one better.

File: include/event_scripts.h (hunk 1 above)
Comment (PikalaxALT) [reply include/event_scripts.h]: These point to text objects in data/event_scripts.s, not data/strings.s.

Comment (jiangzhengwenjz) [reply include/event_scripts.h]: Yes, I'm wrong. But stuff above (gText_WouldYouLikeToSaveTheGame, etc) shouldn't be here

**Pull request merged**